### PR TITLE
feat: update package versions and enhance request body handling

### PR DESCRIPTION
**Pull request by andypost**
- Bump version to 1.35.6 in Cargo.toml and openapi-config.json
- Introduce `KnownSize::into_full_body()` for accurate Content-Length calculation
- Optimize process filtering in `find_unitd_processes()`
- Update dependencies to latest compatible versions
- Add error handling for I/O operations in request body materialization
- Improve test coverage for new features and edge cases

### CODE REVIEW

File: src/nxt_unit.c [hunk 1]
```diff
@@ -1303,6 +1405,103 @@ nxt_unit_process_req_headers(nxt_unit_ctx_t *ctx, nxt_unit_recv_msg_t *recv_msg,
...
+         * (fields_count * sizeof(nxt_unit_field_t)) must lie within the
+         * received buffer before the per-field sptr loop below -- and
+         * before nxt_unit_request_group_dup_fields() and the language
+         * modules -- dereference fields[i].  64-bit math keeps the
+         * multiplication from overflowing a 32-bit fields_count.
+         */
+        if (nxt_slow_path(sizeof(nxt_unit_request_t)
+                          + (uint64_t) vr->fields_count
+                            * sizeof(nxt_unit_field_t)
+                          > vsize))
```
[the remark this answers is not in the document]
Comment (andypost) [reply src/nxt_unit.c:1431]: Declined — false positive. The guard this asks for already exists ~20 lines above in the same function (nxt_unit_process_req_headers): `if (nxt_slow_path(recv_msg->size < sizeof(nxt_unit_request_t))) return NXT_UNIT_ERROR;`. So `vsize >= sizeof(nxt_unit_request_t)` is guaranteed before `vr->fields_count` is read — the OOB read cannot occur — and the existing `sizeof(request) + fields_count*sizeof(field) > vsize` check then bounds the trailing fields[] region.

File: src/nxt_conf_validation.c [hunk 2]
```diff
@@ -3364,9 +3558,17 @@ nxt_conf_vldt_cgroup_path(nxt_conf_validation_t *vldt, nxt_conf_value_t *value,
...
-    sprintf(path, "/%*s/", (int) cgpath.length, cgpath.start);
+    if (cgpath.length == 0
+        || memchr(cgpath.start, '\0', cgpath.length) != NULL)
+    {
+        return nxt_conf_vldt_error(vldt,
+                                   "The cgroup path \"%V\" is invalid.",
+                                   &cgpath);
+    }
+
+    snprintf(path, sizeof(path), "/%.*s/", (int) cgpath.length, cgpath.start);
```
[the remark this answers is not in the document]
Comment (andypost) [reply src/nxt_conf_validation.c:3569]: Declined — false positive. This snprintf cannot truncate: the top of nxt_conf_vldt_cgroup_path rejects `cgpath.length >= NXT_MAX_PATH_LEN - strlen(NXT_CGROUP_ROOT) - 1`, which reserves *more* headroom than the `/%.*s/` buffer needs (cgpath.length + 3 bytes incl. NUL), so the formatted path is always complete and the `/../` strstr runs on the full string. No truncation bypass is reachable.

File: src/nxt_otel.c [hunk 3]
```diff
@@ -182,43 +271,62 @@ nxt_otel_span_add_body(nxt_http_request_t *r)
...
+    // dont bother logging an unset status
+    if (r->status == 0) {
+        return;
+    }
 
-    status_val = (nxt_str_t) {
-        .start  = status_buf,
-        .length = 3,
-    };
+    n = sprintf((char *) status_buf, "%d", (int) r->status);
```
[the remark this answers is not in the document]
Comment (andypost) [reply src/nxt_otel.c]: Good catch — real (status_buf is 8 bytes, an int can format to 11). Fixed in #140: sprintf -> snprintf(status_buf, sizeof(status_buf), ...). Leaving this thread open for you to resolve once #140 lands.

File: src/nxt_controller.c [hunk 4]
```diff
@@ -709,6 +709,77 @@ nxt_runtime_controller_socket(nxt_task_t *task, nxt_runtime_t *rt)
...
+        {
+            nxt_alert(task, "controller: SO_PEERCRED failed %E", nxt_errno);
+            return NXT_ERROR;
+        }
+
+        if (cred.uid != 0 && cred.uid != nxt_euid) {
+            nxt_alert(task, "controller: rejecting connection from uid %d "
+                      "(unitd uid %d); set socket permissions accordingly",
+                      (int) cred.uid, (int) nxt_euid);
+            return NXT_ERROR;
```
[the remark this answers is not in the document]
Comment (andypost) [reply src/nxt_controller.c:745]: Legitimate compat point. Decision: keep the peer-UID gate — it is the intended control-socket boundary (filesystem perms are defense-in-depth) — and document it. #140 adds a note under --control-user/group/mode in unitd.8 making that explicit; a matching CHANGES bullet goes into the release changelog PR (#132). Leaving this open for you.
